### ISSUE TITLE
Change mqtrigger to POST as JSON and not text

### ISSUE THREAD

**Issue by tomwparsons** [fission/fission#275]
We have a function that receives JSON data via a POST request. When we set up a mqtrigger to invoke the function and then send a JSON message to the topic, the function is receiving the JSON data as text (or we think it is).

Can the mqtrigger POST the message as JSON?

**Comment by soamvasani** [comment]
Sorry, didn't mean to close it.

@life1347 thoughts? Maybe add a content type to the trigger spec?

**Comment by life1347** [comment]
How about we can send a JSON message that contains message body and its type, then Trigger will set the HTTP content-type based on the type field?
```
{
   "body": "{\"a\":\"b\"}",
   "type": "json"
}
```

**Comment by dmlond** [comment]
you could make it more generic, using body, headers, params:
```
{
  "body": "{\"a\":\"b"}",
  "headers": { "Content-Type": "application/json", ...},
  "params": {"a": "value a","b": "value b", ...}
}
```

**Comment by soamvasani** [comment]
Adding a type=json information inside a json struct doesn't seem to add much... 

I was thinking the trigger spec can simply have a Content-Type that gets sent to the function call, that's all. The environment and/or function can then interpret the content-type.

**Comment by yqf3139** [comment]
Ran into the same issue. Adding the content-type can help the environment to interpret the content-type automatically.
https://github.com/yqf3139/fission/blob/487348235967b8ffee32662cc72a0d9054f35a5b/mqtrigger/messageQueue/nats.go#L106

**Comment by soamvasani** [comment]
@yqf3139 exactly like that, except instead of hardcoding it would be good to take the value from a new trigger.Spec.ContentType attribute.

**Comment by life1347** [comment]
I'll push a PR to add content-type support :)

**Comment by soamvasani** [comment]
Works on master, will be available in 0.2.1
